File: src/test/java/ar/com/flow/bankaccount/OptionalTransactionRecordAssert.java
package ar.com.flow.bankaccount;

import ar.com.flow.bankaccount.transaction.Receipt;
import org.assertj.core.api.AbstractAssert;
import org.assertj.core.api.Assertions;

import java.util.Optional;

public class OptionalTransactionRecordAssert extends AbstractAssert<OptionalTransactionRecordAssert, Optional<Receipt>> {
    public static OptionalTransactionRecordAssert assertThat(Optional<Receipt> statement) {
        return new OptionalTransactionRecordAssert(statement);
    }
    
    public OptionalTransactionRecordAssert(Optional<Receipt> statement) {
        super(statement, OptionalTransactionRecordAssert.class);
    }

    public TransactionRecordAssert isPresent() {
        Assertions.assertThat(actual).isPresent();

        return new TransactionRecordAssert(actual.get());
    }
}

File: src/test/kotlin/ar/com/flow/bankaccount/OptionalTransactionRecordAssert.kt
package ar.com.flow.bankaccount

import ar.com.flow.bankaccount.transaction.receipt.Receipt
import org.assertj.core.api.AbstractAssert
import org.assertj.core.api.Assertions
import java.util.*

class OptionalTransactionRecordAssert(statement: Optional<Receipt>) :
    AbstractAssert<OptionalTransactionRecordAssert?, Optional<Receipt>>(
        statement,
        OptionalTransactionRecordAssert::class.java
    ) {

    fun isPresent(): ReceiptAssert {
        Assertions.assertThat(actual).isPresent
        return ReceiptAssert(actual.get())
    }

    companion object {
        fun assertThat(statement: Optional<Receipt>): OptionalTransactionRecordAssert {
            return OptionalTransactionRecordAssert(statement)
        }
    }
}